libnetwork/iptables/firewalld.go

```diff
@@ -282,6 +282,22 @@ func DelInterfaceFirewalld(intf string) error {
 	return nil
 }
 
+func SetupInterfaceFirewalldZone(bridgeName string, enable bool) error {
+	// Either add or remove the interface from the firewalld zone
+	if firewalldRunning {
```

Comment:
Perhaps we should move the `firewalldRunning`  check into `AddInterfaceFirewall` and ` DelInterfaceFirewalld` functions

(note to self: I want to get rid of directly using that package-level variable, as it depends on a side-effect of other functions being called, so it's not guaranteed to always be set)

Reply:
If we do, we can inline the `Add..`, `Del...` calls in the code, which (I think) makes it more transparent what we're doing (not having to pass that awkward `insert` boolean);

```go
func setIcc(version iptables.IPVersion, bridgeIface string, iccEnable, insert bool) error {
	iptable := iptables.GetIptable(version)
	var (
		table      = iptables.Filter
		chain      = "FORWARD"
		args       = []string{"-i", bridgeIface, "-o", bridgeIface, "-j"}
		acceptArgs = append(args, "ACCEPT")
		dropArgs   = append(args, "DROP")
	)

	if insert {
		if !iccEnable {
			iptable.Raw(append([]string{"-D", chain}, acceptArgs...)...)

			if !iptable.Exists(table, chain, dropArgs...) {
				if err := iptable.RawCombinedOutput(append([]string{"-A", chain}, dropArgs...)...); err != nil {
					return fmt.Errorf("Unable to prevent intercontainer communication: %s", err.Error())
				}
			}
		} else {
			iptable.Raw(append([]string{"-D", chain}, dropArgs...)...)

			if !iptable.Exists(table, chain, acceptArgs...) {
				if err := iptable.RawCombinedOutput(append([]string{"-I", chain}, acceptArgs...)...); err != nil {
					return fmt.Errorf("Unable to allow intercontainer communication: %s", err.Error())
				}
			}
			// Ensure bridge interface is added to the docker zone if firewalld is running.
			if err := iptables.AddInterfaceFirewalld(bridgeName); err != nil {
				return err
			}
		}
	} else {
		// Remove any ICC rule.
		if !iccEnable {
			if iptable.Exists(table, chain, dropArgs...) {
				iptable.Raw(append([]string{"-D", chain}, dropArgs...)...)
			}
		} else {
			if iptable.Exists(table, chain, acceptArgs...) {
				iptable.Raw(append([]string{"-D", chain}, acceptArgs...)...)
			}
			// Remove bridge interface from the docker zone if firewalld is running.
			if err := iptables.DelInterfaceFirewalld(bridgeName); err != nil {
				return err
			}
		}
	}

	return nil
}
```

Reply:
Agreed, but I have no idea what should be done with `firewalldRunning` check. I believe it might already be in the right place, other functions are using this check